examples/finetuning/dpo_tic_tac_toe/src/dpo_tic_tac_toe/ttc_move_selector_function.py

```
@@ -0,0 +1,242 @@
+                "board": board, "player_symbol": player_symbol
+            },
+            metadata={"turn_index": turn_index},
+        )
+
+        # === TTC Pipeline ===
+
+        # 1. SEARCH: Generate N candidate moves
+        candidate_items = await searcher.ainvoke([initial_item])
+
+        if not candidate_items:
+            raise RuntimeError("No valid candidate moves generated!")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)